### DISCUSSION TITLE
How to use Connect API key with pilot?

### DISCUSSION THREAD

**Discussion by alcarvalho** [fastlane/fastlane#17034] [Q&A] [answered]
Hi there!

I have looked everywhere in the docs and I can't find a way to use API keys instead of username / password for `pilot` or `deliver`. I tried looking at the code and the only place that looks for any env variables is on `Spaceship::ConnectAPI::Token`, but I can't find a way to make sure it's using that on `pilot`. 

Thanks,
Andre

**Comment by joshdholtz** [comment, marked as answer]
Hey! I actually just commented on this on another issue that I should figure out how to document that 😊

I've started working on rework of `Spaceship::ConnectAPI` that will make this easier to use 👇 
https://github.com/fastlane/fastlane/pull/17042

I'm also going to create a new action called something like `spaceship_stats` that will print out the number of request made to each domain (so its clear how much you are actually using the new API 🙃)

**Reply by joshdholtz** [reply]
@alcarvalho Here is a preview of what you will be able to do in next minor version of _fastlane_

## Fastfile changes

```rb
before_all do
  # Use official App Store Connect API with
  # JWT auth when able to
  require 'spaceship'
  Spaceship::ConnectAPI.auth # loads from ENV

  # Could also do
  # Spaceship::ConnectAPI::Client.auth(
  #  key_id: "D797SM4YA8",
  #  issuer_id: "81306a3d-8f1c-4847-8cb3-5b5770adf30b",
  #  filepath: "../AuthKey_D797SM4YA8.p8"
  #)
end

after_all do
  # Print summary of Apple endpoints and auth type requests were made
  spaceship_stats
end
```

### Example output
![Screen Shot 2020-08-13 at 12 01 12 PM](https://user-images.githubusercontent.com/401294/90165976-1f00d000-dd5f-11ea-8d14-81100f91aae2.png)
![Screen Shot 2020-08-13 at 12 03 23 PM](https://user-images.githubusercontent.com/401294/90165981-1f996680-dd5f-11ea-978f-962518daa4e9.png)

**Reply by alcarvalho** [reply]
This is so nice! Thank you!
